### ISSUE TITLE
Not an issue, only annoying in-video ads

### ISSUE THREAD

**Issue by zymase** [gorhill/uBlock#1136]
This is not really the place to expose a quest for a rule but since all ads are handled here and now with my uBo rules and that I encounter one place which has recently made some changes to impose in-video ads, I'll assume my rudeness and pray for pardon (wow!):

Dailymotion.com now carries on in-video ads which appear randomly here. I've tried everything I could, was aware of, nothing does it. My uBo settings are "extreme" : blocks by default 3-rd party / 3-rd party scripts / 3-rd party frames. Authorized the minimum to allow site & video correct dsplays.

I'm not only bothered with these intrusive in-video ads but also puzzled.
Forgot to mention that these video ads appear with a counter (top-right) and that if I click on the ad-video (so as to halt it) it opens a new tab, blank here since the new tab's content must have been blocked by uBo.

Sorry again and thanks (a lot) if anyone can find the correct rule to stop this intruder :)
I won't make it a habit, first & last time, promised!

**Comment by gorhill** [comment]
Does this work for you?

    ||dailymotion.com/ac/video/$image,first-party,redirect=2x2-transparent.png

If it works I will add it to uBO's own filters.

Note: No need to walk on eggs regarding filters, the guideline is simple: if something cannot be blocked by EasyList (due to ABP's limitations) **or** if it can be blocked by EasyList but at the cost of having to create undesirable exception filters (like `@@||pubads.g.doubleclick.net`), I will be happy to look into it, this is what motivated me to extend uBO's capabilities in the first place.

**Comment by zymase** [comment]
Many thanks, gorhill! The rule you've provided above seems to do the job :+1: 
I do say "seems" because as I stated above the in-video ads appear(ed) randomly; I've tested several videos and all rendered nicely as if the intruder was kicked out : no words to express this particular feeling!
OK for the guideline, thanks for the information. Much appreciated.

Off-topic : gorhill, uBlockO is really a beautiful product, the very first add-on I'd install on a new browser. really an everyday companion. And so many users happy, relaxed, unstressed... I cannot close this off-topic without wishing to you, to all, and to uBlockOrigin the Very Best For A Very Happy New Year! (and this enthusiastic sincerity is not a mental consequence of my joy to have this new rule!)

Thanks again.

**Comment by gorhill** [comment]
I get an ad with that video: <http://www.dailymotion.com/video/x3jubr9_people-wiping-out-on-their-hoverboards-is-the-gift-that-keeps-on-giving_news>.

**Comment by zymase** [comment]
I've just tested the video you mention, gorhill, as well, again, several others, and all ad-free.
This disparity may be relevant of the random distribution of in-video ads at Dailymotion,
I'll keep on testing other videos.
I've tested your video with and without a rule I added yesterday,
* mc.dailymotion.com * block
and in both cases, no ads. * mc.dailymotion.com * block because all runs fine without mc.dailymotion.com and that I've been unable to find info about it, so for the time being I've blocked it, but doesn't seem to be pertinent in this scenario.

**Comment by gorhill** [comment]
Hmm I am unable to reproduce the ad, I think I may have not been using the filter above after I added it to uBO's filter list and remove it from my custom filter.

**Comment by zymase** [comment]
That would explain it all. Anyway, if I meet the in-video ads again I'll post it here if the topic is still opened.
Thanks, gorhill.

**Comment by zymase** [comment]
Not sure if I should post here since the thread is closed or open a new issue ... sorry if I mistake. I'm really not a techie guy....

The rule mentioned above, if it seemed to work for some time, appears now nonfunctional.
Dailymotion has found a very ingenious way to impose its in-video ads, as before they are injected randomly and difficult to reproduce on one link. On the homesite as well as embedded. A true pain. I just cannot and will not accept to be force-feeded for the sake of a video, but on another hand several sites (mainly here in France) decided to choose Dailymotion as a partner and therefor avoiding Dailymotion means missing those embedded videos.

I truly hope a rule will ba able to circumvent this nasty in-video advertisement.

**Comment by gorhill** [comment]
I need a specific URL where you see an ad.

**Comment by zymase** [comment]
I have a site with embedded Dailymotion : http://www.franceinfo.fr/
It displays the in-video ad almost systematically, but not always. Under there are several other videos, all handled by Dailymotion and, viewing one after another leads more likely to one carrying an in-video ad.
The vicious thing here is the random distribution.

Thanks, gorhill

**Comment by gorhill** [comment]
These are Flash-based players. This is a live-stream of a radio station, there is no way ads can be blocked, there are not separated from the content itself.

**Comment by zymase** [comment]
Flash is removed here, only HTML5, and if indeed the above site's introduction page is that of radio (audio only) I nevertheless have it play an in-video ad (randomly) before getting to the audio only radio ... just tested it again, with the video ad appearing. Maybe an HTML5 specificity?

The site I mentioned, which has  Dailymotion as partner, has a dedicated video page: http://www.franceinfo.fr/videos

About Flash and HTML5, many sites don't call HTML5 if they notice the user's browser runs Flash ...

**Comment by gorhill** [comment]
There is a thread on ABP forum about this: <https://adblockplus.org/forum/viewtopic.php?f=2&t=43430&p=145703>.

The ABP guy says this works fine with EasyList and Liste FR. Can you confirm?

**Comment by zymase** [comment]
Thanks for the ABP Forum link, I'm on my way to have a look there.

About your dailymotion video, gorhill: I've just viewed without any ad. That's what I was saying, it's totally random. I've viewed this afternoon 10-12 videos with no ads, then suddenly one with an in-video ad....
Anyway, not a drama. Fortunately there is an 'add-escape button top-right. Without that ad-exit I would not bear whatever ad for the sake of whatever video. Drives me insane (almost!).

Thanks.

**Comment by gorhill** [comment]
When you get a video with an ad, can you tell me if you see this script in the logger:

    static1.dmcdn.net/playerv5/1f2ad7e2d3ce1a215d74.dmp.js

If so, blocking this script prevents the ad from appearing on my side -- though this causes a delay before the real video starts playing, but I could remediate this on my side.

**Comment by zymase** [comment]
I've just tested and was lucky (relativity!) to get an in-video ad!!!

uBO Logger -> http://www.dailymotion.com/video/x3lp2li_la-france-battue-en-finale-du-tqo_sport
Back to Logger which indeed displayed:
http://static1.dmcdn.net/playerv5/1f2ad7e2d3ce1a215d74.dmp.js

Grrr, could that be related to the crime?!

Thanks, gorhill, for your investigations.

**Comment by zymase** [comment]
Now that I want an ad video I cannot get any!!! Totally random. Just ran 15 videos (few seconds each), and no ad! Even if I block the script I'll won't know (how would I know before either an ad or 48 hrs without?).

Can I ask you, gorhill, how to write the rule for blocking static1.dmcdn.net/playerv5/1f2ad7e2d3ce1a215d74.dmp.js since I can not blovk it from the Logger- Thanks

**Comment by gorhill** [comment]
Try

    ||static1.dmcdn.net/playerv5/1f2ad7e2d3ce1a215d74.dmp.js$script

**Comment by zymase** [comment]
OK, gorhill, that's what I did before your answer (I ignore what I know!)
So:
I installed the rule ||static1.dmcdn.net/playerv5/1f2ad7e2d3ce1a215d74.dmp.js$script
I ran again http://www.dailymotion.com/video/x3lp2li_la-france-battue-en-finale-du-tqo_sport
Video took some time to arrive, as in your experiment.
Ran the Logger and indeed the script appeared as blocked.

Now, even if this script is the culprit, I'm wondering if the name is not random as well
When I see scripts such as 1f2ad7e2d3ce1a215d74.dmp.js  I tend to believe the 1f2ad7e2d3ce1a215d74 part is random ...

**Comment by gorhill** [comment]
> I'm wondering if the name is not random as well

That's why I am investigating more before making a filter out of this.

**Comment by zymase** [comment]
I ran several videos from Dailymotion with the blocking rule for the mentioned script, and all videos 1- had no ad, 2- were delayed.
Either the delay performed on all videos (with & without ad) either the ad was repeated when the algorithm noted the ad had not been issued, which would mean at least 2 or 3 Dailymotion ad processes:
1- Random
2- Maybe on certain videos and not on all (though I've experienced a same video with and without)
3- Maybe a policy regarding the IP-identified user, his behavior regarding the ad and so on ...

No idea, but terrifically tough to find a guiding line in what seems to be a multi-factor well organized ad-distribution system.

**Comment by gorhill** [comment]
I created a filter which appears to work -- it does not rely on the filename. See if this works fine on your side.

**Comment by zymase** [comment]
I'm not at all aware of the way github is organized, off its methodology, so I landed on this rule which I believe is the correct one, and added it to my filters:
dailymotion.com##script:inject(1136#issuecomment-170371863.js)
and of course removed previous ||static1.dmcdn.net/playerv5/1f2ad7e2d3ce1a215d74.dmp.js$script

I tested the rule with several embedded as well as home Dailymotion videos and have had NO in-video ads make their way. I hope I installed the correct rule, otherwise it would be luck!

I think I did install the right rule because I notice a perhaps 1-2 second delay before the video start.
May sound funny but for users such as myself code is a universe and github a labyrinth so please pardon my wandering.

Of course I hope the medicine will appear more as surgery than as plain pills. 
I admire and your commitment and your talent, gorhill, and flattery is not in my habits.

Can I ask you one last (presumably) thing, gorhill? How does this rule function. Does it call a scipt here at github to afterwards inject it in the page? You know, besides what I manage to do with uBO because it has been elaborated to be easily handled by the users even non-techie, I understand 0+ of 99+ reality behind the scene!

Thanks A MILLION, gorhill. Should I encounter an issue in this counter-strike that I'd pass the word here, of course. But seeing what you've achieved I'm more than reasonably confident.

What a nice feeling, that of being no longer force-fed!

**Comment by gorhill** [comment]
For the rule to work, you need an up to date version of `resources.txt`, the rule causes a scriptlet to be loaded on the page which defuse the video player layer.

**Comment by zymase** [comment]
I noticed on my uBO '3rd-party filters' that an update for 'uBlock filters' was available, so I've updated.
Now I see that indeed the rule dailymotion.com##script:inject(1136#issuecomment-170371863.js) appears in the uBlock filters list.
Hence, removing this rule from 'My filters'.

I'm learning, thanks for teaching! What I lack as well is the terminology, not only the procedures.

**Comment by zymase** [comment]
Another site with Dailymotion embedded videos showing in-video ads:
http://www.20minutes.fr/tv/t-as-vu/214905-un-bebe-phoque-nait-au-zoo-de-brookfield-et-emeut-les-internautes

I hate to have to be the one bringing this information... gosh, what a pain.

**Comment by gorhill** [comment]
> Another site with Dailymotion embedded videos showing in-video ads.

I can't see ad on this site. Using a French server through VPN, still no ads for whatever video I tried. What are your other settings?

**Comment by zymase** [comment]
I just returned to the 20minutes.fr page I mentioned and this time no ads. Like previously if random ads injection is the reason.

My settings, which ones?

uBO 1.5.3 / Firefox 43.0.4 x64 / Windows 7SP1 x64
Flash is not installed, neither as a browser plug-in neither as an activex

uBO is set to block 3rd-party frames/scripts/access
Concerning DailyMotion I have set in My rules:
* dailymotion.com * noop
* dm.gg * noop
* dmcdn.net * noop
* hapyak.dmcdn.net * block
* mc.dailymotion.com * block
* tofu.dmcdn.net * block

Nothing in My filters concerning either dailymotion.net / dm.gg / dmcdn.net.
67 262 network filters ＋ 65 574 cosmetic filters at this time.
3rd-party filters : All uBO filters except Unbreak
Updates are manual, once a day every morning.

Maybe a Firefox setting? Ask me anything related which could seem pertinent.

**Comment by jjacquess** [comment]
@gorhill Adguards approach: https://github.com/AdguardTeam/AdguardForWindows/issues/591
http://forum.adguard.com/showthread.php?7285-Resolved-spiegel-de-video-ads/page2

they can block (pre) flash ads - many sites use flash (exclusively) only for one reason: it is more difficult to block ads - disadvantageous regarding HTML5 penetration

**Comment by gorhill** [comment]
@jjacquess Spiegel video ads are blocked by uBO, not an issue, and what you link to is AdGuard for Windows, it's not an extension.

@zymase The whole time I was testing with Chromium, this works well. I tested with Firefox 43, and for some reasons which I still have to figure, the scriptlet is not injected in the Daily Motion frame, so that explains why there is an issue.

**Comment by gorhill** [comment]
@zymase I confirm there was a regression bug in 1.5.3 which explains the issue you are experiencing with Firefox. Fixed in 5591ff34f1a47adca847d52979ce98f64663c16f and published the fix on AMO in 1.5.4.

**Comment by zymase** [comment]
uBO has just been updated to ver.1.5.5 and testing the Dailymotion in-video ads on the site itself as embedded elsewhere has shown up to now zero ads.
I'll never repeat it enough: your work and commitment, gorhill, is exceptional. And if investigating on an issue (which is rather a new advertisement trick) wasn't enough, the search & development must moreover be handled with several browsers in mind.
Be thanked, sincerely.

**Comment by zymase** [comment]
In-video ads are back on Dailymotion despite uBlock dedicated filters:
||dailymotion.com/ac/video/$image,first-party,redirect=2x2-transparent.png
dailymotion.com##script:inject(1136.js)

On the site itself and of course on Dailymotion embedded videos.
This is driving users nuts because the ad appears on every video even if, as before, blockable after 5 seconds. But 5 seconds for each video drives you crazy. I'm bound to blacklist DailyMotion, just can't stand it anymore.

Thanks again for uBlockO whatever the issue with this damned Dailymotion site. Insane ad policy.

**Comment by gorhill** [comment]
> On the site itself and of course on Dailymotion embedded videos

I've tried many videos, and haven't been able to see an ad so far. Do you have a URL from where you consistently get an ad?

**Comment by zymase** [comment]
Hi gorhill,

I've just tested now, first available video on Dailymotion.com/fr and another, embedded:

http://www.dailymotion.com/video/x3x3v63_orlm-221-2p-design-et-finition-gs7-aussi-bien-que-l-iphone_tech

http://videos.leparisien.fr/video/chelsea-psg-paris-sauve-le-foot-francais-10-03-2016-x3wupw5

Both include the in-video ad as mentioned.

It seems here to be no longer random as I get them systematically, anyway since last night when I went to have a look at Dailymotion to compare with my surprise to see again the ad on a dailymotion embedded video (leparisien.fr).

I do hope this is not again the vicious random ad display issue because then attempting to find the solution would be double trouble.

Thanks again for your concern. I seldom visit dailymotion.com, the most bothering are embedded videos as several French sites use dailymotion for their video delivery ...

**Comment by gorhill** [comment]
> http://www.dailymotion.com/video/x3x3v63_orlm-221-2p-design-et-finition-gs7-aussi-bien-que-l-iphone_tech

Thanks, I got an ad with this one.

**Comment by zymase** [comment]
Not with the embedded one on leparisien.fr if I understand correctly.  Odd, again & again.

**Comment by gorhill** [comment]
I found clearing the cookies for Daily Motion I systematically get the ad -- not necessarily the case if I do not clear the cookies.

**Comment by zymase** [comment]
OK- I use a Firefox add-on called 'Self-Destructing Cookies' which removes a site's cookies once the site closed unless this site is set as an exception for keep. But the cookie is not touched as long as the user is on the site, in which case the cookie is handled following the user's default policy for cookies.

My cookie default policy is for session only (hence, all cookies for session only unless exception rule to keep or block).

Now, Dailymotion here has no exception rule, hence its cookie is session-only -> Surprise: if I set an exception rule (keep) for dailymotion.com then ... the in-video ad doesn't reappear as you've experienced, same if I quit dailymotion since the cookie is 1- not removed, 2- is not a session cookie.

Seems dailymotion wants not only a session cookie but a "long-term" cookie ...

Looks, looked ok... until -- in this scheme -- I tested another video on dailymotion and had -- again -- an in-line ad:
http://www.dailymotion.com/video/x3x3qxy_marie-gillain-agacee-au-grand-journal-quand-maitena-biraben-lui-montre-sa-une-du-magazine-lui_tv

Now get to understand the morbid, cynical algorithm they have and i'll eat my hat (or beret).

mama mia ...

**Comment by gorhill** [comment]
Could try the following filter by pasting it in your _My filters_ pane in the dashboard?

    ||liverail.com/js/liverail.admanager$script,important,domain=dailymotion.com

Tell me if this helps.

**Comment by zymase** [comment]
Just added the new filter, unsuccessful (old cookie scheme here, that is session cookie).

**Comment by gorhill** [comment]
Remove the filter I asked to test above, then update filter lists -- see if this fixes the issue.

**Comment by zymase** [comment]
gorhill: you are a genius! For he's a jolly good fellow, for he's a jolly good fellow!

Fantastic. I removed first testing new filter, updated filter lists, checked a video on dailymotion.com then an embedded one on leparisien.fr and ... NO IN-VIDEO ADS! (I'm not yelling, I'm happy!).

Many many thanks, gorhill. I don't know how you do it, and do it so fast, but you sure do it :)

Nice, nice, nice, thanks again!

**Comment by ffreed** [comment]
Just to thank you for your job.
The new dailymotion filter works perfectly.

**Comment by brunob** [comment]
Hi, and first a huge thank for your work on this addon :) I got problem to read another dailymotion video here : 

http://www.dailymotion.com/video/x3wwy33_loi-travail-alain-deneault-explique-pourquoi-notre-societe-est-devenue-folle_news

I've purged and updated all my active lists and here is what i got in my dynamic filter tab (my personnal filters are empty) : 

```
* disqus.com * block
* facebook.com * block
* facebook.net * block
* googletagservices.com * block
* taboola.com * block
* twitter.com * block
behind-the-scene * 3p noop
behind-the-scene * 3p-frame noop
facebook.com facebook.com * noop
facebook.com facebook.net * noop
twitter.com twitter.com * noop
```
I can provide an export of my config if necessary.

**Comment by IDKwhattoputhere** [comment]
I can play the video fine with your filters. There must be something else on your side that breaks it.

**Comment by gorhill** [comment]
Purge one of uBO's filter list than force an update. See if this works.

**Comment by zymase** [comment]
What I've noticed very recently is that DailyMotion has again modified its code regarding in-video ads in such a way that when latest uBlock Origin's dedicated blocking is active the video simply doesn't start at all.

I've found an interesting script over at Greasyfork that eradicates totally the in-video ad by replacing DailyMotion's video player with the browser's native HTML5 player. Works fine here on Firefox 45.0.1 

**Dailymotion Raw Html5 Player** at https://greasyfork.org/en/scripts/18056-dailymotion-raw-html5-player

I'm afraid Dailymotion is stubborn to the point of loosing users rather than ads.

**Comment by zymase** [comment]
Latest **_uBlockO filters_** (_uBlock filters_ in _3-rd party filters_) fixes once again the continuously changing Dailymotion in-video code.
Thanks again, gorhill.

**Comment by brunob** [comment]
\o/ @gorhill thx a lot it works :)

**Comment by umairkhalil** [comment]
I am experiencing this same issue with the video ad at the start of the video on dailymotion. I am running v1.6.8 and happens both on chrome and firefox. I updated the lists but no effect. I am thinking that dailymotion found a way around your filter.

**Comment by gorhill** [comment]
URL?

**Comment by zymase** [comment]
A fast one : http://www.dailymotion.com/video/x47xvjt_actube-du-30-04-le-tube-du-30-04_tv

**Comment by petmue** [comment]
> URL?
> A fast one : http://www.dailymotion.com/video/x47xvjt_actube-du-30-04-le-tube-du-30-04_tv

It's not always triggered, but I had the the Firefox debugger running and found it loads for example this advertising video before it plays the actual video:

http://vid2.ec.dmcdn.net/sec%28f440e69f9435fe5d7eab6f2b1a4b1d06%29/frag%2877%29/775/319/2838903577_mp4_h264_aac_uhd.ts

**Comment by gorhill** [comment]
> http://www.dailymotion.com/video/x47xvjt_actube-du-30-04-le-tube-du-30-04_tv

I am able to reproduce, using a VPN to FR. Will investigate.

**Comment by gorhill** [comment]
> It's not always triggered

Clearing cookies for the site before forcing a page reload made the ad appears reliably on my side.
